librz/core/rop.c

```diff
@@ -0,0 +1,1248 @@
+	if (!gadget_info || !reg_info) {
+		return;
+	}
+
+	RzRegInfo *reg_info_dup = rz_reg_info_dup(reg_info);
+	if (!reg_info_dup) {
+		return;
+	}
+	switch (evt->type) {
+	case RZ_IL_EVENT_MEM_READ: {
+		const RzILEventMemRead *mem_read = &evt->data.mem_read;
+		reg_info->is_mem_read = true;
+		reg_info->is_mem_write = false;
+		reg_info->is_var_write = false;
+		reg_info_dup->new_val = rz_bv_to_ut64(mem_read->address);
+		break;
+	}
+	case RZ_IL_EVENT_MEM_WRITE: {
+		reg_info->is_mem_write = true;
+		reg_info->is_mem_read = false;
+		reg_info->is_var_write = false;
+		const RzILEventMemWrite *mem_write = &evt->data.mem_write;
+		reg_info_dup->init_val = rz_bv_to_ut64(mem_write->old_value);
+		reg_info_dup->new_val = rz_bv_to_ut64(mem_write->new_value);
+		break;
+	}
+	case RZ_IL_EVENT_VAR_WRITE: {
+		reg_info->is_var_write = true;
+		reg_info->is_mem_read = false;
+		reg_info->is_mem_write = false;
+		const RzILEventVarWrite *var_write = &evt->data.var_write;
+		RzBitVector *init_val = rz_il_value_to_bv(var_write->old_value);
+		RzBitVector *new_val = rz_il_value_to_bv(var_write->new_value);
+		if (!init_val || !new_val) {
+			rz_bv_free(init_val);
+			rz_bv_free(new_val);
+			break;
+		}
+		// reg_info_dup->init_val = rz_bv_to_ut64(init_val);
+		reg_info_dup->new_val = rz_bv_to_ut64(new_val);
+		if (is_stack_pointer(core, reg_info->name)) {
+			gadget_info->stack_change += rz_bv_to_ut64(new_val) - reg_info->new_val;
+		}
+		rz_bv_free(init_val);
+		rz_bv_free(new_val);
+		break;
+	}
+	default:
+		break;
+	}
+	rz_list_append(gadget_info->dependencies, reg_info_dup);
+}
+
+static void fill_rop_gadget_info_from_events(RzCore *core, RzRopGadgetInfo *gadget_info, const RzILEvent *curr_event,
+	RzILEvent *event, RzPVector /*<RzILEvent *>*/ *vec, const bool is_dependency) {
+	if (!gadget_info) {
+		return;
+	}
+	switch (event->type) {
+	case RZ_IL_EVENT_VAR_READ: {
+		RzILEventVarRead *var_read = &event->data.var_read;
+		RzRegInfo *reg_info = rz_rop_gadget_info_get_modified_register(gadget_info, var_read->variable);
+		if (reg_info && !is_dependency) {
+			RzRegInfo *new_reg_info = rz_reg_info_dup(reg_info);
+			if (!new_reg_info) {
+				break;
+			}
+			RzBitVector *val = rz_il_value_to_bv(var_read->value);
+			if (!val) {
+				break;
+			}
+			new_reg_info->new_val = rz_bv_to_ut64(val);
+			rz_rop_gadget_info_update_register(gadget_info, new_reg_info);
+			rz_reg_info_free(new_reg_info);
+			rz_pvector_push(vec, event);
+			rz_bv_free(val);
+			break;
+		}
+		if (is_dependency && curr_event) {
+			// shouldn't take this path if it is a dependency
+			if (curr_event->type == RZ_IL_EVENT_VAR_READ) {
+				break;
+			}
+			rz_rop_gadget_info_add_dependency(core, gadget_info, curr_event, reg_info);
+			break;
+		}
+		if (reg_info) {
+			break;
+		}
+		RzBitVector *val = rz_il_value_to_bv(var_read->value);
+		if (!val) {
+			break;
+		}
+		reg_info = rz_reg_info_new(core, event, rz_bv_to_ut64(val), rz_bv_to_ut64(val));
```

Comment:
define a `ut64 val64 = rz_bv_to_ut64(val);` then pass `val64` as argument for `rz_reg_info_new`